src/amenities/amenities_server/Controllers/AmenitiesController.cs

```diff
@@ -0,0 +1,98 @@
+using Microsoft.AspNetCore.Mvc;
+
+namespace amenities_server.Controllers;
+
+[ApiController]
+[Route("amenities")]
+public class AmenitiesController : ControllerBase
+{
+    private IAmenityPersistence _amenityPersistence;
+
+    public AmenitiesController()
+    {
+        _amenityPersistence = Services.GetAmenityPersistence();
+    }
+
+    //get: /amenities
+    [HttpGet]
+    public IActionResult GetAmenities()
+    {
+        var amenities = _amenityPersistence.GetAmenities();
+
+        if (amenities == null)
+        {
+            return NotFound();
+        }
+
+        return Ok(amenities);
+    }
+
+    //get: /amenities/{id}
+    [HttpGet("{id}")]
+    public ActionResult<Amenity> GetAmenityByID(int id)
+    {
+        var amenity = _amenityPersistence.GetAmenityByID(id);
+
+        if (amenity == null)
+        {
+            return NotFound();
+        }
+
+        return amenity;
+    }
+
+    //delete: /amenities/{id}
+    [HttpDelete("{id}")]
+    public IActionResult DeleteAmenity(int id)
+    {
+        //validate session
```

Comment:
NO ACTION REQUIRED
It might be worthwhile to create a validator interface and class with a ValidatePermission(permissionName, sessionKey)
function, that can be used across all your controllers (not super applicable to this service, but would be a nice template for others maybe... even better if we have some way to share that resource... idk)